python/pyiceberg/expressions/literals.py

```diff
@@ -65,6 +66,9 @@ class Literal(Generic[L], ABC):
     def __init__(self, value: L, value_type: Type[L]):
         if value is None or not isinstance(value, value_type):
             raise TypeError(f"Invalid literal value: {value!r} (not a {value_type})")
+        if isinstance(value, float):
```

Comment:
Python has lazy evaluation, so we can combine these in the same line 👍🏻 
```suggestion
        if isinstance(value, float) and isnan(value):
```

Reply:
Fixed! I never realized that -- good to know.